feat(tui): use DA1 response to determine OSC 52 support

gpanders opened the pull request:
Many terminals now include support for OSC 52 in their Primary Device Attributes (DA1) response. This is preferable to using XTGETTCAP because DA1 is _much_ more broadly supported.

Closes: https://github.com/neovim/neovim/issues/34472

gpanders committed:
feat(tui): use DA1 response to determine OSC 52 support

Many terminals now include support for OSC 52 in their Primary Device
Attributes (DA1) response. This is preferable to using XTGETTCAP because
DA1 is _much_ more broadly supported.

gpanders commented:
Once terminal support for DA1 is rolled out more broadly (i.e. in stable releases) we will deprecate/remove using XTGETTCAP. As far as I'm aware, there are no terminals that support XTGETTCAP but do _not_ also include OSC 52 support in their DA1 response  (according to https://github.com/neovim/neovim/issues/34472#issuecomment-2994298500).

gpanders commented:
> > Once terminal support for DA1 is rolled out more broadly (i.e. in stable releases) we will deprecate/remove using XTGETTCAP.
> 
> That seems to contradict "DA1 is much more broadly supported" ? imo let's just remove it now 

It's supported in nightly/unreleased builds. But this hasn't made its way to most users yet.

> would dropping XTGETTCAP happen to fix https://github.com/neovim/neovim/issues/34816 ?

Yea, altho we still use XTGETTCAP to determine support for 24 bit colors, so that could still trigger it in the right circumstances.

j4james commented on src/nvim/vterm/state.c:
```diff
@@ -17,6 +17,11 @@
 
 #define strneq(a, b, n) (strncmp(a, b, n) == 0)
 
+// Primary Device Attributes (DA1) response.
+// We make this a global (extern) variable so that we can override it with FFI
+// in tests.
+char vterm_primary_device_attr[] = "1;2;52";
```
FYI, the `1;2` response is a device-specific format that predates DEC's standardization of the `DA` response. The second parameter is actually a bitfield value that represents extensions specific to the VT100 range of devices. It isn't expected to have additional parameters after it.

If you want to indicate a level-1 device using the _standard_ `DA` format, the first parameter should be `61`, and then any additional parameters represent the supported extensions. For example, a level-1 device with just the new clipboard extension would be `61;52`.

Probably not super important, though, since many apps that parse the `DA` response will likely ignore the first parameter anyway.

gpanders committed:
refactor(vterm): update vterm DA1 response

Update vterm's DA1 response to the more modern version that indicates
level 1 support for VT100 emulation (61) as well as ANSI color support
(22).

gpanders commented on runtime/plugin/osc52.lua:
```diff
@@ -19,33 +19,77 @@ vim.api.nvim_create_autocmd('UIEnter', {
+    -- Clear existing OSC 52 value, since this is a new UI we might be attached to a different
+    -- terminal
+    do
       local termfeatures = vim.g.termfeatures or {} ---@type TermFeatures
```
Not sure. I feel like we need both, because a user may change their clipboard setting at any time and knowing whether we support OSC 52 independent of that is useful.

If anything I'd like to remove `g:termfeatures` in favor of a more generalized "UI features/capabilities" API call.
